tests/std/tests/P0355R7_calendars_and_time_zones_time_zones/test.cpp

```diff
@@ -91,9 +117,268 @@ void timezone_names_test() {
+    assert_local(tz, danger_begin - days{2}, local_info::unique, first, empty); // two days before
+    assert_local(tz, danger_begin - hours{1}, local_info::unique, first, empty); // one hour before
+    assert_local(tz, danger_begin, result, first, second); // danger begin
+    assert_local(tz, danger_begin + minutes{30}, result, first, second); // danger middle
+    assert_local(tz, danger_end, local_info::unique, second, empty); // danger end
+    assert_local(tz, danger_end + hours{2}, local_info::unique, second, empty); // one hour after
+    assert_local(tz, danger_end + days{2}, local_info::unique, second, empty); // two days after
+}
+
+void timezone_local_info_test() {
+    const auto& tzdb = get_tzdb();
+    {
+        // positive offset (UTC+10/+11) can fall in previous transition
+        using namespace Sydney;
+        auto tz = tzdb.locate_zone(Tz_name);
+        assert(tz != nullptr);
+        validate_get_local_info(tz, Day_to_Std, local_info::ambiguous);
+        validate_get_local_info(tz, Std_to_Day, local_info::nonexistent);
+    }
+    {
+        // negative offset (UTC-8/-7) can fall in next transition
+        using namespace LA;
+        auto tz = tzdb.locate_zone(Tz_name);
+        assert(tz != nullptr);
+        validate_get_local_info(tz, Day_to_Std, local_info::ambiguous);
+        validate_get_local_info(tz, Std_to_Day, local_info::nonexistent);
+    }
+}
+
+template <typename _Dur>
+void validate_precision(const time_zone* tz, const pair<Transition, Transition>& transition_pair, _Dur precision) {
+    const auto& first       = transition_pair.first;
+    const auto& second      = transition_pair.second;
+    const auto transition   = first.end<_Dur>();
+    const auto danger_begin = get_danger_begin<_Dur>(first, second);
+    const auto danger_end   = get_danger_end<_Dur>(first, second);
+    sys_info first_info     = tz->get_info(first.begin<_Dur>());
+    sys_info second_info    = tz->get_info(second.begin<_Dur>());
+
+    // test correct transition is picked
+    assert(tz->get_info(transition) == second_info);
+    assert(tz->get_info(transition - precision) == first_info);
+
+    // test ambiguous/nonexistent info is handled
+    assert(tz->get_info(danger_end).result == local_info::unique); // exact end of danger zone
+    assert(tz->get_info(danger_end - precision).result != local_info::unique); // just inside danger zone
+    assert(tz->get_info(danger_begin).result != local_info::unique); // exact start of danger zone
+    assert(tz->get_info(danger_begin - precision).result == local_info::unique); // just before danger zone
+
+    // test precision is not lost when converting to local
+    assert(tz->to_local(transition) == second.local_begin<_Dur>());
+    assert(tz->to_local(transition + precision) == second.local_begin<_Dur>() + precision);
+    assert(tz->to_local(transition - precision) == first.local_end<_Dur>() - precision);
+
+    // test precision is not lost when converting to local

```

Comment:
This comment `converting to local` is repeated, but it looks like this test is `to_sys`.